docs/cqrs/query/index.md

```diff
@@ -0,0 +1,58 @@
+# Query
+
+Query is just an class that implements the `IQuery<TResult>` interface (there's also non-generic `IQuery` interface but it shouldn't be used directly). The only generic parameter specifies the type that the query returns when executed. It should be a DTO (because most of the time it will be serialized). Queries get the data from the system but don't modify it.
+
+## Contract
+
+Consider the following query
+
+```csharp
+public class ProjectInfoDTO
+{
+    public string ProjectId { get; set; }
+    public string Name { get; set; }
+}
+
+[AllowUnauthorized]
+public class FindProjectsMatchingName : IQuery<List<ProjectInfoDTO>>
```

Comment:
Query's name should describe the result and means of getting it e.g. `ProjectNameById`, emphasizing the read-side of CQRS. `Find` shouldn't be used as we would like to set an example here.

It would be also worth to add a short paragraph describing naming convention.